pkg/gateway/client/identity.go

```diff
@@ -400,13 +401,17 @@ func (c *Client) RemoveIdentityAndUser(ctx context.Context, id *types.Identity)
 			return err
 		}
 
-		// Attempt to delete the user
+		if err := userQuery.First(&user).Error; err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+			return err
+		}
```

Comment:
Nit: return earlier if err is ErrRecordNotFound?

Reply:
Done.